panel/src/components/Layout/Overlay.vue

```diff
@@ -51,16 +51,26 @@ export default {
       scrollTop: 0
     };
   },
+  destroyed() {
+    // make sure scrolling is enabled again
+    // when overlay is destroyed without closing it
+    this.close(true);
+  },
   methods: {
-    close() {
+    close(destroyed = false) {
       // it makes it run once
       if (this.isOpen === false) {
         return;
       }
 
       this.isOpen = false;
-      this.$emit("close");
-      this.restoreScrollPosition();
+
+      // no need for event and restoring scroll
+      // when closed due to navigating away from view
+      if (destroyed === false) {
+        this.$emit("close");
```

Comment:
What could be listening for the `close` event? Could there be a case where the listening component needs to know about the closed overlay even when the component was destroyed?

Reply:
When testing my experience was the other way around: if the event was emitted, the form drawer was listening to it and wanted to focus some input that didn't exist anymore either. I think, the overlay really only gets destroyed without being closed when a whole view changes or similar, in which case all other components get destroyed as well. 

Or we would need to add more if statements to the listening components to make sure that their reactions might not be always possible anymore.

Reply:
To be honest I don't grasp the whole event setup around the `Overlay`. As far as I can tell, the only components that listen to the `close` event directly are `Dialog` and `Drawer`. But since events bubble up to the topmost component, I'm not sure about possible edge cases. It probably doesn't even make sense that this `close` event bubbles up, maybe we should explicitly stop it in the direct parent component. Given that `Dialog` and `Drawer` both emit another `close` event in their handlers for `Overlay`, it's all even more complicated. :(